src/Shape/Box.js

```diff
@@ -84,33 +106,31 @@ export default class Box extends BaseClass {
       else if (mode[1] === "extent") d.bottom = min(values);
       else if (typeof mode[1] === "number") d.bottom = max([min(values), quantile(values, mode[1] / 100)]);
 
+      // Compute data for outliers.
+      values.forEach(value => {
```

Comment:
A couple of notes for creating the outliers:
* Instead of looping through the `values` array, loop through `d.values` so that you can have access to the data point (for the `data` and `i` keys)
* No need to construct `dataObj` for values that are not outliers. Calculate the value (`const value = this._value(d, i)` and then check if it is an outlier before creating the object (`if (value < d.bottom || value > d.top)`)
* Make sure to set `data` and `i` to the actual data point and index from the user's data (here, you are setting each `data` object to the nest data)